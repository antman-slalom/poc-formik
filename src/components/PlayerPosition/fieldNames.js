const POSITION = 'position';
const INTERNATIONAL = 'international';
const ON_LOAN = 'onLoan';
const INJURED = 'injured';
const STARTER = 'starter';

export default {
    POSITION,
    INTERNATIONAL,
    ON_LOAN,
    INJURED,
    STARTER,
};
