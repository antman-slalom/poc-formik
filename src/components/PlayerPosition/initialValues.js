import fieldsNames from './fieldNames';

export default {
    [fieldsNames.POSITION]: '',
    [fieldsNames.INTERNATIONAL]: '',
    [fieldsNames.ON_LOAN]: '',
    [fieldsNames.INJURED]: '',
    [fieldsNames.STARTER]: '',
};
