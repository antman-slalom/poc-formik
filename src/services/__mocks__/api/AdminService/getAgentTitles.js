import ServiceHelper from '../../ServiceMockHelper';

const createObject = (titleId, titleName, isDTM, marginSourceId, quotaSourceId) => ({
    titleId,
    titleName,
    isDTM,
    marginSourceId,
    quotaSourceId,
});

const successData = [
    createObject(0, 'Atlanta United', true, 1, 1),
    createObject(1, 'Atlanta Falcons', false, 1, 2),
    createObject(2, 'Atlanta Braves', true, 2, 2),
    createObject(3, 'Atlanta Hawks', true, 3, 2),
    createObject(4, 'Atlanta Knights', true, 2, 1),
    createObject(5, 'GA Tech', true, 1, 3),
    createObject(6, 'Georgia State', false, 1, 1),
    createObject(7, 'University of Georgia', false, 3, 3),
];

const success = ServiceHelper.createOkResponse(successData);

const unauthorized = ServiceHelper.createUnauthorizedResponse();

const serverError = ServiceHelper.createUnauthorizedResponse();

const throwError = ServiceHelper.createReject();

export default {
    success,
    unauthorized,
    serverError,
    throwError,
};
