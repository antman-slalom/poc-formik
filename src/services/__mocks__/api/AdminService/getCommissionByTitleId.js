import ServiceHelper from '../../ServiceMockHelper';

const createObject = (agentTitleCommissionId, commission, dollarThreshold, percentThreshold) => ({
    agentTitleCommissionId,
    commission,
    dollarThreshold,
    percentThreshold,
});

const successData = [
    createObject(0, 100, 1000, 0.15),
    createObject(1, 200, 2000, 0.25),
    createObject(2, 300, 3000, 0.35),
    createObject(3, 400, 4000, 0.45),
];

const success = ServiceHelper.createOkResponse(successData);

const unauthorized = ServiceHelper.createUnauthorizedResponse();

const serverError = ServiceHelper.createUnauthorizedResponse();

const throwError = ServiceHelper.createReject();

export default {
    success,
    unauthorized,
    serverError,
    throwError,
};
