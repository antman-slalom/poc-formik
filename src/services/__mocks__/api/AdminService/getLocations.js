import ServiceHelper from '../../ServiceMockHelper';

const createObject = (id, name, address = undefined, city, state, zip = undefined) => ({
    id,
    name,
    address,
    city,
    state,
    zip,
});

const successData = [
    createObject(1, 'Allianz Field', null, 'Saint Paul', 'Minnesota'),
    createObject(2, 'Audi Field', null, 'Washington', 'D.C.'),
    createObject(3, 'Avaya Stadium', null, 'San Jose', 'California'),
    createObject(4, 'Banc of California Stadium', null, 'Los Angeles'),
    createObject(5, 'BBVA Stadium', null, 'Houston'),
    createObject(6, 'BC Place', null, 'Vancouver'),
    createObject(7, 'BMO Field', null, 'Toronto'),
    createObject(8, 'CenturyLink Field', null, 'Seattle'),
    createObject(9, 'Children\'s Mercy Park', null, 'Kansas City', 'Kansas'),
    createObject(10, 'Dick\'s Sporting Goods Park', null, 'Commerce City', 'Colorado'),
    createObject(11, 'Dignity Health Sports Park', null, 'Carson', 'California'),
    createObject(12, 'Exploria Stadium', null, 'Orlando', 'Florida'),
    createObject(13, 'Gillette Stadium', null, 'Foxborough', 'Massachusetts'),
    createObject(14, 'Mapfre Stadium', null, 'Columbus', 'Ohio'),
    createObject(15, 'Mercedes-Benz Stadium', null, 'Atlanta'),
    createObject(16, 'Nippert Stadium', null, 'Cincinnati'),
    createObject(17, 'Providence Park', null, 'Portland', 'Oregon'),
    createObject(18, 'Red Bull Arena', null, 'Harrison', 'New Jersey'),
    createObject(19, 'Rio Tinto Stadium', null, 'Sandy', 'Utah'),
    createObject(20, 'Saputo Stadium', null, 'Montreal'),
    createObject(21, 'SeatGeek Stadium', null, 'Bridgeview', 'Illinois'),
    createObject(22, 'Talen Energy Stadium', null, 'Chester', 'Pennsylvania'),
    createObject(23, 'Toyota Stadium', null, 'Frisco', 'Texas'),
    createObject(24, 'Yankee Stadium', null, 'Bronx', 'New York'),
];


const success = ServiceHelper.createOkResponse(successData);

const unauthorized = ServiceHelper.createUnauthorizedResponse();

const serverError = ServiceHelper.createUnauthorizedResponse();

const throwError = ServiceHelper.createReject();

export default {
    success,
    unauthorized,
    serverError,
    throwError,
};
