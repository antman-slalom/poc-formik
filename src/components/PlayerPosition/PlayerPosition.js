import React from 'react';
import Grid from '@material-ui/core/Grid';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Radio from '@material-ui/core/Radio';

import Checkbox from '../../formik/Checkbox';
import RadioGroup from '../../formik/RadioGroup';
import { withNamespace } from '../../formik/utils';
import Switch from '../../formik/Switch';
import isEqual from 'react-fast-compare';

import { Typography } from '@material-ui/core';
import fieldNames from './fieldNames';

class PlayerPosition extends React.Component {
    shouldComponentUpdate(nextProps) {
        return !isEqual(nextProps, this.props);
    }

    render() {
        const {
            classes,
            namespace,
        } = this.props;

        return (
            <React.Fragment>
                <Typography variant='h5'>Position: </Typography>
                <Grid className={ classes.Grid } container>
                    <Grid className={ classes.GridItem } item md={ 4 } xs={ 12 }>
                        <RadioGroup name={ withNamespace(namespace, fieldNames.POSITION) } >
                            <FormControlLabel control={ <Radio /> } label='Goalkeeper' value='goalkeeper' />
                            <FormControlLabel control={ <Radio /> } label='Fullback' value='fullback' />
                            <FormControlLabel control={ <Radio /> } label='Back' value='back' />
                            <FormControlLabel control={ <Radio /> } label='Midfielder' value='midfielder' />
                            <FormControlLabel control={ <Radio /> } label='Striker' value='striker' />
                        </RadioGroup>

                    </Grid>
                    <Grid className={ classes.GridItem } item md={ 4 } xs={ 12 }>
                        <Checkbox label='International' name={ withNamespace(namespace, fieldNames.INTERNATIONAL) } onBlur={ this.handleBlur } onChange={ this.handleChange } variant='outlined' />
                        <br/>
                        <Checkbox label='On Loan' name={ withNamespace(namespace, fieldNames.ON_LOAN) } onBlur={ this.handleBlur } onChange={ this.handleChange } variant='outlined' />

                    </Grid>
                    <Grid className={ classes.GridItem } item md={ 4 } xs={ 12 }>
                        <Switch label='Injured' name={ withNamespace(namespace, fieldNames.INJURED) } onBlur={ this.handleBlur } onChange={ this.handleChange }/>
                        <br/>
                        <Switch label='Starter' name={ withNamespace(namespace, fieldNames.STARTER) } onBlur={ this.handleBlur } onChange={ this.handleChange }/>
                    </Grid>
                </Grid>
            </React.Fragment>
        );
    }
}

export default PlayerPosition;
