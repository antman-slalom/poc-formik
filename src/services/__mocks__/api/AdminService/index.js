import getAgentTitles from './getAgentTitles';
import getCommissionByTitleId from './getCommissionByTitleId';
import getLocations from './getLocations';
import getPlayer from './getPlayer';
import getStadiumInfo from './getStadiumInfo';
import getQuotasByTitleId from './getQuotasByTitleId';
import saveAgentTitle from './saveAgentTitle';

const AdminService = {};

// get calls
AdminService.getAgentTitles = () => getAgentTitles.success;
AdminService.getCommissionByTitleId = () => getCommissionByTitleId.success;
AdminService.getLocations = () => getLocations.success;
AdminService.getPlayer = id => getPlayer.success(id);
AdminService.getStadiumInfo = id => getStadiumInfo.success(id);
AdminService.getQuotasByTitleId = () => getQuotasByTitleId.success;

// save calls
AdminService.saveAgentTitle = () => saveAgentTitle.success;

export default AdminService;
