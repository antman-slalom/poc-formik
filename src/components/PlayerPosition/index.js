import { withStyles } from '@material-ui/styles';
import PlayerPosition from './PlayerPosition';

import styles from './styles';

export default withStyles(styles)(PlayerPosition);
