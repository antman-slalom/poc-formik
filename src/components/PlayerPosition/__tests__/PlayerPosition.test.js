import React from 'react';
import { shallow } from 'enzyme';

import PlayerPosition from '../PlayerPosition';

// need to have relative path in order for Jest to pull the mocked service
import AdminService from '../../../services/AdminService';

// let jest know to pull mock data from __mocks__
jest.mock('../../../services/AdminService');

describe('PlayerPosition', () => {
    let wrapper;
    let instance;

    const props = {
        classes: {},
        namespace: 'TEST',
    };

    beforeEach(() => {
        AdminService.getLocations = jest.fn(() => Promise.resolve({ data: [] }));

        wrapper = shallow(<PlayerPosition { ...props } />);
        instance = wrapper.instance();
    });

    afterEach(() => {
        // clears all mocks to prevent mock bleeding from test to test,
        jest.clearAllMocks();

        // mock the API service after the clear
        jest.mock('../../../services/AdminService');
    });

    // =================================
    // Render and Snapshot
    // =================================

    it('matches snapshot', () => {
        expect(wrapper.instance()).toMatchSnapshot();
    });

    it('should render', () => {
        expect(wrapper.instance()).not.toBeNull();
    });

    // =================================
    // Components call correct functions
    // =================================

    // N/A

    // =================================
    // Functions act as expected
    // =================================
    it('Test shouldComponentUpdate to be false', () => {
        const result = instance.shouldComponentUpdate(props, wrapper.state());
        expect(result).toBe(false);
    });

    it('Test shouldComponentUpdate to be true', () => {
        const result = instance.shouldComponentUpdate({}, {});
        expect(result).toBe(true);
    });
});
