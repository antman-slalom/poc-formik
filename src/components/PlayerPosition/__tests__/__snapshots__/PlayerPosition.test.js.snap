// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPosition matches snapshot 1`] = `
PlayerPosition {
  "context": Object {},
  "props": Object {
    "classes": Object {},
    "namespace": "TEST",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <PlayerPosition
        classes={Object {}}
        namespace="TEST"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <React.Fragment>
        <ForwardRef(WithStyles)
          variant="h5"
        >
          Position: 
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          container={true}
        >
          <ForwardRef(WithStyles)
            item={true}
            md={4}
            xs={12}
          >
            <RadioGroup
              name="TEST.position"
            >
              <ForwardRef(WithStyles)
                control={<ForwardRef(WithStyles) />}
                label="Gaolkeeper"
                value="gaolkeeper"
              />
              <ForwardRef(WithStyles)
                control={<ForwardRef(WithStyles) />}
                label="Fullback"
                value="fullback"
              />
              <ForwardRef(WithStyles)
                control={<ForwardRef(WithStyles) />}
                label="Back"
                value="back"
              />
              <ForwardRef(WithStyles)
                control={<ForwardRef(WithStyles) />}
                label="Midfielder"
                value="midfielder"
              />
              <ForwardRef(WithStyles)
                control={<ForwardRef(WithStyles) />}
                label="Striker"
                value="striker"
              />
            </RadioGroup>
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            item={true}
            md={4}
            xs={12}
          >
            <CheckboxWithLabel
              label="International"
              name="TEST.international"
              variant="outlined"
            />
            <br />
            <CheckboxWithLabel
              label="On Loan"
              name="TEST.onLoan"
              variant="outlined"
            />
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            item={true}
            md={4}
            xs={12}
          >
            <Switch
              label="Injured"
              name="TEST.injured"
            />
            <br />
            <Switch
              label="Starter"
              name="TEST.starter"
            />
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
